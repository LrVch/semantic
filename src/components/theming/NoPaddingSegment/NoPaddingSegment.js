import {
  Segment
} from 'semantic-ui-react'
import styled from 'styled-components'

export default styled(Segment)`
  &&& {
    /* padding: 0; */
    border-color: ${({theme}) => theme.second};
  }
`