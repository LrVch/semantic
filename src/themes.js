export default {
  tomato: {
    name: 'tomato',
    main: 'tomato',
    second: '#ff6347ab',
    invert: 'white',
    dark: 'rgba(0,0,0,.87)'
  },
  palevioletred: {
    name: 'tomato',
    main: 'palevioletred',
    second: '#db709394',
    invert: 'white',
    dark: 'rgba(0,0,0,.87)'
  }
}