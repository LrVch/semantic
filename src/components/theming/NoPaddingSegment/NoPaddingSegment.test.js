import 'jest-styled-components'

import { configure, mount, shallow } from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'
import NoPaddingSegment from './NoPaddingSegment'
import React from 'react'
import { mountWithTheme } from '../../../test'
import styled from 'styled-components'
import themes from '../../../themes'

configure({ adapter: new Adapter() })

describe('ThemedDivider', () => {

  let wrapper
  let mounted

  beforeEach(() => {
    wrapper  = shallow(<NoPaddingSegment />)
    mounted = mountWithTheme(<NoPaddingSegment />, themes.tomato)
  })

  test('should create', () => {
    expect(wrapper).toBeTruthy()
  })

  test('should has "border-color" equals to "#ff6347ab" ', () => {
    expect(mounted).toMatchSnapshot()
    expect(mounted).toHaveStyleRule('border-color', '#ff6347ab', {
      modifier: '&&&'
    })
  })
})


