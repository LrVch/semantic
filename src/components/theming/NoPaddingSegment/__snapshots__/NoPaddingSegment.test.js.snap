// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedDivider should has "border-color" equals to "#ff6347ab"  1`] = `
.c0.c0.c0 {
  border-color: #ff6347ab;
}

<Styled(Segment)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "Group": [Function],
        "Inline": [Function],
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  &&& {
    /* padding: 0; */
    border-color: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "Styled(Segment)",
        "foldedComponentIds": Array [],
        "handledProps": Array [
          "as",
          "attached",
          "basic",
          "children",
          "circular",
          "className",
          "clearing",
          "color",
          "compact",
          "content",
          "disabled",
          "floated",
          "inverted",
          "loading",
          "padded",
          "piled",
          "placeholder",
          "raised",
          "secondary",
          "size",
          "stacked",
          "tertiary",
          "textAlign",
          "vertical",
        ],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <Segment
      className="c0"
    >
      <div
        className="ui segment c0"
      />
    </Segment>
  </StyledComponent>
</Styled(Segment)>
`;
